src/Microsoft.Azure.WebJobs.Extensions.Kafka/Diagnostics/BatchEventActivityProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.Diagnostics
+{
+    internal class BatchEventActivityProvider : ActivityProvider, IDisposable
+    {
+        private List<ActivityLink> activityLinks;
+        private readonly IKafkaEventData[] kafkaEvents;
+        private static string KafkaBatchTriggerActivityName { get; } = "MultipleKafkaTrigger.Process";
+
+        public BatchEventActivityProvider(IKafkaEventData[] kafkaEvents, string consumerGroup) : base(kafkaEvents[0].Topic, consumerGroup)
+        {
+            this.kafkaEvents = kafkaEvents;
+            this.activityLinks = new List<ActivityLink>();
+            this.CreateAndStartActivity();
+        }
+
+        public void CreateAndStartActivity()
+        {
+            if (ActivitySource.HasListeners())
+            {
+                this.CreateActivityLinksForAllEvents();
+                this.CreateActivity(KafkaBatchTriggerActivityName, ActivityKind.Consumer, null, activityLinks);
+                this.AddActivityTags();
+                this.StartActivity();
+            }
+        }
+
+        public void CreateActivityLink(string traceParentId)
+        {
+            var traceParentFields = traceParentId.Split('-');
+            if (traceParentFields.Length != 4)
+            {
+                //ERROR: Invalid traceparent Header
```

Comment:
error case is missed here

Reply:
I have changed the implementation with a method provided by ActivityContext which checks if the traceparent is valid or not.